### PR TITLE
HPA update its status even if getting metrics failed

### LINKED ISSUE #21542: HPA should update its status even if metrics getting fails

### PR DESCRIPTION
**Pull request by piosz**
fix #21542

cc @mwielgus

### CODE REVIEW

File: pkg/controller/podautoscaler/horizontal_test.go [hunk 1]
```diff
@@ -98,6 +98,10 @@ func (tc *testCase) prepareTestClient(t *testing.T) *fake.Clientset {
 						MinReplicas: &tc.minReplicas,
 						MaxReplicas: tc.maxReplicas,
 					},
```
Comment (mwielgus) [inline pkg/controller/podautoscaler/horizontal_test.go]: Please add a test that checks that current replicas is updated on error.

**Comment by mwielgus** [comment]
LGTM, just one request for extra tests.

File: pkg/controller/podautoscaler/horizontal_test.go (hunk 1 above)
Comment (piosz) [inline pkg/controller/podautoscaler/horizontal_test.go]: Done

**Comment by fgrzadkowski** [comment]
This fixes a blocker for 1.2.

**Comment by mwielgus** [comment]
Merging manually per @davidopp request.

**Pull request merged**